js/src/logger.ts

```diff
@@ -3215,6 +3279,36 @@ export class SpanImpl implements Span {
     let sanitizedAndInternalData = { ...internalData };
     mergeDicts(sanitizedAndInternalData, sanitized);
 
+    const serializableInternalData: typeof sanitizedAndInternalData = {};
```

Comment:
@manugoyal

Reply:
Also since this function is getting quite complicated, it'd be nice to separate into smaller sub-functions, or use scoping to indicate the lifetime of variables. Like maybe we can write it as
```
const [serializableInternalData, lazyInternalData] = (() => {
})();
```

because `sanitizedAndInternalData` is not used elsewhere.

Reply:
Changed the type signatures, skipped the refactoring (open to doing that at the end, perhaps).

Reply:
I think refactoring would be nice. The function is getting quite large and complicated. Assuming it's covered by tests, there's minimal risk.